### DISCUSSION TITLE
Cannot find Microsoft.CodeAnalysis.dll -- for some people in the same build system

### DISCUSSION THREAD

**Discussion by nschuessler** [dotnet/roslyn#55933] [Q&A]
**Version Used**: 

Roslyn 3.9, Visual Studio 16.9.6

**Steps to Reproduce**:

1.  Install our custom analyzer
2.  Do a build.
3.  You (on some machines) get the following error:
```
CSC : error CS8032: An instance of analyzer <redacted>Analyzer cannot be created from F:\src\packages\<redactedanalyzer>\3.21.519.2\analyzers\dotnet\cs\redactedanalyzer.dll : Could not load file or assembly 'Microsoft.CodeAnalysis, Version=3.9.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. The system cannot find the file specified.. [F:\<redacted>\redacted.csproj]
```

**Expected Behavior**:

Since we have Visual Studio 16.9 and Roslyn 3.9, it should find it right?

**Actual Behavior**:

We get the cannot find Microsoft.CodeAnalysis.dll error.

The strange this is that once it happens, it seems to always happen, even if you revert back to a previous version of the custom analyzer. So there seems to be a cache somewhere that remembers a bad state.

This does not seem to get fixed by:
1) Stamping Roslyn 3.9 on the dependencies of the analyzer nuget
2) Clearing .nuget package cache and rebuilding
3) Reverting to an earlier working version of the custom analyzer.
4) `dotnet --version` doesn't seem to be relevant, as the same version install can or cannot show the result (repros under dotnet 3.1.409)

Q: How can I debug this? 
Q: Are there other caches to clear?

Note: the custom analyzer package does not have Microsoft.CodeAnalysis.dll included in the nuget. The documentation seems to indicate they are to be excluded.

We cannot have customers use or analyzers if it randomly fails. Blocking issue.

**Comment by ghost** [comment]
I couldn't figure out the best area label to add to this issue. If you have write-permissions please help me learn by adding exactly one area label.

**Comment by nschuessler** [comment]
It seems part of the issue is when you build a .Net standard analyzer with .Net 50 and then try to run with `dotnet build` under .Net Core 3.1.  I will try to debug why this matters.

**Comment by Youssef1313** [comment]
@nschuessler Analyzers should target .NET Standard. They cannot target .NET 5 (or .NET Core or .NET Framework)

**Comment by nschuessler** [comment]
I did target .Net standard, but there may be some dependency that got picked up that ties them to .Net 5 along the way.
i.e. Roslyn 3.9?

**Comment by Youssef1313** [comment]
It's expected to work with 3.9. So you may wait for someone from the compilers team to better assist then.
